### PR TITLE
Initial implementation

### PR DESCRIPTION
**Pull request by nikita-vanyasin**
Provides the implementation of HTTP streaming to allow port-forwarding
from firewalled server to the Platform.

The implementation is based on Kubernetes project source code, specifically [k8s.io/apimachinery/pkg/util/httpstream](https://pkg.go.dev/k8s.io/apimachinery/pkg/util/httpstream) package.
Main changes are:
- support for port forwarding in reverse direction
- all k8s apimachinery and dependencies removed
- no klog logging - the logging is delegated to client of this package

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+		$(shell zutano go mod replacements)
+	go get \
+		$(shell zutano go mod latest \
+			github.com/arangodb-managed/apis \
+			github.com/arangodb-managed/apis-helper \
+			github.com/arangodb-managed/arangodb-helper \
+			github.com/arangodb-managed/internal-apis \
+			github.com/arangodb-managed/log-helper \
+			github.com/arangodb-managed/metrics-helper \
+			github.com/arangodb-managed/refs-helper \
```
Comment (Robert-Stam) [inline Makefile]: please remove from list what isn't needed/accessible, in a public package this should contain public refs only.

File: README.md [hunk 2]
```diff
@@ -1 +1,14 @@
 # portforward-helper
+
+Port forward helper for ArangoGraph Insights Platform.
+
+Provides the implementation of HTTP streaming to allow port-forwarding
```
Comment (Robert-Stam) [inline README.md]: HTTP --> HTTP(S)

**Review by Robert-Stam — COMMENTED** [review]
Q: Instead of copying all this code in an own package (which leads to maintaining it), can't we import the code from the k8s project itself instead?
If not, LGTM (small remark only about makefile)

File: README.md (hunk 2 above)
Comment (nikita-vanyasin) [reply README.md]: done

File: Makefile (hunk 1 above)
Comment (nikita-vanyasin) [reply Makefile]: done

**Comment by nikita-vanyasin** [comment]
> Q: Instead of copying all this code in an own package (which leads to maintaining it), can't we import the code from the k8s project itself instead? If not, LGTM (small remark only about makefile)

Surely was thinking about it. But there are few problems:
1) I had to adjust k8s implementation a bit to support forwarding in reverse direction. There was no option to handle new HTTP stream on server side.
2) k8s impl has many references to other k8s staff, like apimachinery API, meta API package and other. That does not make sense to have as a dependency for migration service.
3) k8s impl uses klog which will result in two types of logging output by the services who use portforward-helper.

**Review by Robert-Stam — APPROVED** [review]
LGTM

**Review by Robert-Stam — APPROVED** [review]
LGTM

**Pull request merged**